src/System.Data.SqlClient/src/System.Data.SqlClient.csproj

```diff
@@ -265,6 +265,12 @@
     </Compile>
     <Compile Include="System\Data\SqlClient\SqlFileStream.Windows.cs" />
   </ItemGroup>
+  <ItemGroup Condition="'$(TargetsNetCoreApp)' == 'true'">
+    <Compile Include="System\Data\SqlClient\SNI\SNIPacket.NetCoreApp.cs" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetsNetCoreApp)' == 'false'">
```

Comment:
This should be '$(TargetsNetCoreApp)' != 'true'

The build system doesn't set an explicit `false` it simply doesn't set the value to true.